### PR TITLE
Removed selfdestruct from BreakInvariantBounty

### LINKED ISSUE #1384: Remove selfdestruct from BreakInvariantBounty
`selfdestruct` can lead to tricky situations, has been the cause of major bugs, and has a debatable usefulness. We should avoid using it whenever possible.

Note that this is not a breaking change since this contract is in the `drafts` directory in the 2.0 release.

### PR DESCRIPTION
**Pull request by Aniket-Engg**
Fixes #1384 

Removed `selfdestruct` from BreakInvariantBounty and related test cases

### CODE REVIEW

**Comment by nventuro** [comment]
Hi @Aniket-Engg, thanks for promptly tackling this! However, we should just remove the function - we should have a function with the same functionality: transferring all balance to the owner, and not letting the bounty continue to be used.

**Comment by Aniket-Engg** [comment]
```
` function abortBounty() public onlyOwner {
     _asyncTransfer(owner(), address(this).balance);
    _claimed = true;
  }`
```
Should it be something like this? @nventuro

**Comment by nventuro** [comment]
That's a good starting point. A couple ideas:
 * Let's rename `_claimed` to `ongoing`, now that a bounty may not be claimable due to the owner canceling it.
 * I think `cancel` is a better verb choice than `abort` in this context, since there's no ongoing process to abort. We should also emit a `Canceled` event.
 * We should add a `require` in `cancelBounty` so that it can only be canceled once.

Let me know what you think!

**Comment by nventuro** [comment]
Currently, a bounty may be not claimed, claimed, or destroyed. A cancelled bounty is obviously not claimed, so we need a better word. I suggested 'ongoing', but 'claimable' may be better (a canceled bounty is not claimable).

**Comment by Aniket-Engg** [comment]
`claimable` looks fine

**Comment by Aniket-Engg** [comment]
@nventuro Please check the changes and let me know your views. I will add tests for `cancelBounty` once it is final.

File: contracts/bounties/BreakInvariantBounty.sol [hunk 1]
```diff
@@ -8,24 +8,25 @@ import "../ownership/Ownable.sol";
...
 
   /**
-   * @dev Determine if the bounty was claimed.
-   * @return true if the bounty was claimed, false otherwise.
+   * @dev Determine if the bounty is claimable.
+   * @return false if the bounty was claimed, true otherwise.
    */
-  function claimed() public view returns(bool) {
-    return _claimed;
+  function isClaimable() public view returns(bool) {
```
Comment (nventuro) [inline contracts/bounties/BreakInvariantBounty.sol]: I'd rename this to simply `claimable`

File: contracts/bounties/BreakInvariantBounty.sol [hunk 2]
```diff
@@ -8,24 +8,25 @@ import "../ownership/Ownable.sol";
  * @dev This bounty will pay out to a researcher if they break invariant logic of the contract.
  */
 contract BreakInvariantBounty is PullPayment, Ownable {
-  bool private _claimed;
+  bool private _claimable = true;
   mapping(address => address) private _researchers;
 
   event TargetCreated(address createdAddress);
+  event BountyCanceled(address bountyAddress);
```
Comment (nventuro) [inline contracts/bounties/BreakInvariantBounty.sol]: I believe this address parameter is not needed, since it will always be the address of the contract.

Comment (Aniket-Engg) [reply contracts/bounties/BreakInvariantBounty.sol]: So, it will be a no parameter event?

File: contracts/bounties/BreakInvariantBounty.sol (hunk 1 above)
Comment (Aniket-Engg) [reply contracts/bounties/BreakInvariantBounty.sol]: will do that

File: contracts/bounties/BreakInvariantBounty.sol (hunk 2 above)
Comment (nventuro) [reply contracts/bounties/BreakInvariantBounty.sol]: Correct!

Comment (Aniket-Engg) [reply contracts/bounties/BreakInvariantBounty.sol]: done

File: contracts/bounties/BreakInvariantBounty.sol (hunk 1 above)
Comment (Aniket-Engg) [reply contracts/bounties/BreakInvariantBounty.sol]: done

**Comment by Aniket-Engg** [comment]
I am also loving being a part of it. Thanks for your appreciation. :blush:

**Pull request merged**